internal/middleware/chain_test.go

```diff
@@ -0,0 +1,99 @@
+//go:build all
+// +build all
+
+package middleware
+
+import (
+	"io"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+)
+
+func TestChain(t *testing.T) {
+	type args struct {
+		endpoint    http.HandlerFunc
+		middlewares []Middleware
+	}
+	tests := []struct {
+		name     string
+		args     args
+		expected string
+	}{
+		{
+			name: "1.2.3",
+			args: args{
+				endpoint: func(writer http.ResponseWriter, request *http.Request) {
+					writer.Write([]byte("3"))
+				},
+				middlewares: []Middleware{
+					func(h http.Handler) http.HandlerFunc {
+						return func(w http.ResponseWriter, r *http.Request) {
+							w.Write([]byte("2."))
+							h.ServeHTTP(w, r)
+						}
+					},
+					func(h http.Handler) http.HandlerFunc {
+						return func(w http.ResponseWriter, r *http.Request) {
+							_, err := w.Write([]byte("1."))
+							if err != nil {
+								println(err)
+							}
+							h.ServeHTTP(w, r)
+						}
+					},
+				},
+			},
+			expected: "1.2.3",
+		},
+		{
+			name: "3.2.1",
```

Comment:
This test is functionally and logically the same as the example I provided for. You need to test whether your chain of functions applies changes in the correct order, not just change "change equation to fit the result".

Reply:
then, only 1 test case is enough?

Reply:
got you wrong in the meet, added other cases

Reply:
fixed